cmd/really/subscription.cpp

```diff
@@ -1,59 +1,63 @@
 #include "subscription.h"
 
-
-Subscriptions::Subscriptions() {
+Subscriptions::Subscriptions()
+{
   subscriptions.resize(129 /* 128 + 1 to handle zero and 128 */);
 }
 
-void Subscriptions::add(const quicr::Name& name, const int len, const Remote& remote ) {
+void
+Subscriptions::add(const quicr::Name& name, uint8_t len, const Remote& remote)
+{
 
-  quicr::Namespace prefix (name, len);
+  const auto prefix = quicr::Namespace{ name, len };
 
-  auto mapPtr =  subscriptions[len].find(prefix.name());
-  if ( mapPtr == subscriptions[len].end() ) {
+  auto mapPtr = subscriptions[len].find(prefix.name());
+  if (mapPtr == subscriptions[len].end()) {
     std::set<Remote> list;
-    list.insert( remote );
-    std::pair<quicr::Name,std::set<Remote>> pair;
-    pair = make_pair( prefix.name() , list );
-    subscriptions[len].insert(pair );
-  }
-  else {
-    std::set<Remote>& list = mapPtr->second;
-    if ( list.find( remote ) == list.end() ) {
-      list.insert( remote ); // TODO - rethink if list is right thing here 
+    list.insert(remote);
+    std::pair<quicr::Name, std::set<Remote>> pair;
+    pair = make_pair(prefix.name(), list);
+    subscriptions[len].insert(pair);
+  } else {
+    auto& list = mapPtr->second;
```

Comment:
Here is an example of where `auto` made the code harder to read. Whereas one would have previously known the type, now one has to go digging into a map to figure it out.

Reply:
And when we refactor to change the type of the map, we'll have to go change a dozen unnecessary type declarations.

Reply:
Yeah! And, good! What is "second" here? I cannot tell with that `auto` there.  If we change the type of whatever `.second` is, we should be forced to inspect downstream impacts.